link-machine-and-node.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+source utils.sh
+
+set -x
+set -e
+
+machine="$1"
+node="$2"
+
+if [ -z "$machine" -o -z "$node" ]; then
+    echo "Usage: $0 MACHINE NODE"
+    exit 1
+fi
+
+uid=$(echo $node | cut -f1 -d':')
+node_name=$(echo $node | cut -f2 -d':')
+
+# BEGIN Hack #260
+# Hack workaround for openshift-metalkube/dev-scripts#260 until it's done automatically
+# Also see https://github.com/metalkube/cluster-api-provider-baremetal/issues/49
+oc --config ocp/auth/kubeconfig proxy &
+proxy_pid=$!
+
+HOST_PROXY_API_PATH="http://localhost:8001/apis/metal3.io/v1alpha1/namespaces/openshift-machine-api/baremetalhosts"
+
+wait_for_json oc_proxy "${HOST_PROXY_API_PATH}" 10 -H "Accept: application/json" -H "Content-Type: application/json"
+
+addresses=$(oc --config ocp/auth/kubeconfig get node -n openshift-machine-api ${node_name} -o json | jq -c '.status.addresses')
+
+machine_data=$(oc --config ocp/auth/kubeconfig get machine -n openshift-machine-api -o json ${machine})
+host=$(echo "$machine_data" | jq '.metadata.annotations["metal3.io/BareMetalHost"]' | cut -f2 -d/ | sed 's/"//g')
+
+if [ -z "$host" ]; then
+    echo "Machine $machine is not linked to a host yet." 1>&2
+    exit 1
+fi
+
+# The address structure on the host doesn't match the node, so extract
+# the values we want into separate variables so we can build the patch
+# we need.
+hostname=$(echo "${addresses}" | jq '.[] | select(. | .type == "Hostname") | .address' | sed 's/"//g')
+ipaddr=$(echo "${addresses}" | jq '.[] | select(. | .type == "InternalIP") | .address' | sed 's/"//g')
+
+host_patch='
+{
+  "status": {
```

Comment:
I'd say to add `"poweredOn": true,` to also mark the hosts as on (otherwise they say powered off)

Reply:
Ironic will update that field when they are adopted correctly.